### ISSUE TITLE
Exception with prayer times [Bug]

### ISSUE THREAD

**Issue by k0T0z** [iamriajul/adhan-dart#46]
======== Exception caught by widgets library =======================================================
The following FormatException was thrown building HomeContentPrayerTimeItem(dirty):
ishaAngle's value isn't define

**Describe the bug**
when choosing "umm alqura uni, mekka" as a calculation method with "hanafi" as a madhab and "based on angle" as a high latitude rule the plugin returns an exception, this isn't supposed to happen.

**To Reproduce**
1. Try to read prayer times after setting calculation parameters as shown above.

**Expected behavior**
The plugin must not return this exception

**Screenshots**
![image](https://user-images.githubusercontent.com/74428638/214915749-c72a50da-0063-4eef-b559-30365aeed6a5.png)
![image](https://user-images.githubusercontent.com/74428638/214915851-f6a6eeb7-8eba-472e-8e25-343731dfc972.png)

**Environment(please complete the following information):**
 - Dart Language Version: 2.18.5
 - Flutter Version: 3.3.9
 - Adhan-Dart Version: 2.0.0-nullsafety.2

**Comment by saifkandil** [comment]
@iamriajul @KarimMohamed20 ?

**Comment by JMApps** [comment]
<img width="313" alt="Снимок экрана 2023-01-31 в 07 50 28" src="https://user-images.githubusercontent.com/40570649/215668331-433986de-34b1-4b85-b4c0-0af8435907a5.png">

Works fine for me when selecting Mecca with prayer time calculation Umm al-Qura. You may not be initializing everything in the correct sequence

**Comment by saifkandil** [comment]
@JMApps the problem is not with selecting "umm alqura uni, mekka"alone try to select it with other parameters I specified above, please

**Comment by JMApps** [comment]
@saifkandil Works fine for me and doesn't throw any exception, Hanafi method for asr prayer is selected in the screenshot and Umm al-Qura for prayer times. But I didn't understand "based on angle".

**Comment by saifkandil** [comment]
@JMApps Sorry I didn't clarify it at first it's twilight angle for high latitude rule (I used it as Based on angle so that user can understand if correctly)
```dart
/// Rules for dealing with Fajr and Isha at places with high latitudes
enum HighLatitudeRule {
  /// Fajr will never be earlier than the middle of the night, and Isha will never be later than
  /// the middle of the night.
  middle_of_the_night,

  /// Fajr will never be earlier than the beginning of the last seventh of the night, and Isha will
  /// never be later than the end of hte first seventh of the night.
  seventh_of_the_night,

  /// Similar to [HighLatitudeRule.seventh_of_the_night], but instead of 1/7th, the faction
  /// of the night used is fajrAngle / 60 and ishaAngle/60.
  twilight_angle
}
```

In short, steps to reproduce is: set your ``High latitude rule`` -> ``twilight_angle``
``Calculation method`` -> ``Umm Al-qura University, Makkah``
``Madhab`` -> ``Hanafi``

**Comment by k0T0z** [comment]
did you managed to reproduce it? @JMApps

**Comment by JMApps** [comment]
Yes brother, I apologize for the late response. I reproduced it, you need to initialize: **twilight_angle** at the very end:

_prayerTimes = PrayerTimes.today(
      _coordinates,
      _prayerParams,
      utcOffset: AppStyles.calculationUtcOffset[timeOffsetIndex],
    );

    _sunnahTimes = SunnahTimes(_prayerTimes);
    _qibla = Qibla(_coordinates);
    _prayerParams.highLatitudeRule = HighLatitudeRule.twilight_angle;
